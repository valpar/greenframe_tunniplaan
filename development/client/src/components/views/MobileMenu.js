import SideModal from "../UI/Modal/SideModal";
import { useState } from "react";

const MobileMenu = (props) => {
  //const [showUserOptions, setShowUserOptions] = useState(false);
  const {
    onClose,
    loginHandler,
    loginMessage,
    loginInfo,
    userPicture,
    userAdminHandler,
    showMobileMenu,
    onUsersManagement,
    usersListOpen,
    admin,
  } = props;

  /*
  const loginHandler = () => {
    setShowUserOptions(true);
  };

  const userRollHandler = (e) => {
    setShowUserOptions(false);
    props.userRollHandler(e);
    onClose();
  };
*/
  const usersManagementHandler = () => {
    onUsersManagement();
    onClose();
  };

  return (
    <SideModal onHidden={showMobileMenu}>
      <div className={`flex flex-col`}>
        <div className="px-4 pt-8">
      {/* {!showUserOptions && ( */}
            <div className="flex flex-col space-y-8 w-full pb-10">
              {/* User photo */}
              <div className="mx-auto w-24 h-24">
                <img
                  src={userPicture}
                  alt="User"
                  className="w-full h-full rounded-full object-cover
            "
                ></img>
              </div>

              <p className="uppercase text-center">
                {loginInfo?.user?.firstName
                  ? loginInfo?.user?.firstName
                  : ""}
              </p>

              <button
                  onClick={loginHandler}
                  className="btn-m-menu"
                  type="button"
                  name="Login Google"
                >
                  {loginMessage}
              </button>
             

              <a
                className="btn-m-menu text-center"
                target="_blank"
                rel="noopener noreferrer"
                href="https://start.hk.tlu.ee/sahtelbeta/sahtel/index.php"
              >
                Sahtel
              </a>
              <a
                className="btn-m-menu text-center"
                target="_blank"
                rel="noopener noreferrer"
                href="https://images.pexels.com/photos/159711/books-bookstore-book-reading-159711.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
              >
                Riiul
              </a>
            </div>
   {/*    )}  */}
        </div>
      </div>
    </SideModal>
  );
};

export default MobileMenu;
