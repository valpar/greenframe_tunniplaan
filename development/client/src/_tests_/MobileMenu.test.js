import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MobileMenu from '../components/views/MobileMenu';

// kui seda ei tee (createPortal), siis target "is not DOM element"
jest.mock('react-dom', () => ({
    ...jest.requireActual('react-dom'),
    createPortal: (element, target) => element, 
  }));

describe('MobileMenu', () => {
  it("matches snapshot", () => {
    const tree = renderer.create(< MobileMenu />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //praegu kommenteerin välja
  /* it('renders Sahtel and Riiul', () => {
    render(<MobileMenu/>);
    expect(screen.getByText('Sahtel')).toBeInTheDocument();
    expect(screen.getByText('Riiul')).toBeInTheDocument();
  });
  it('renders login button when not logged in', () => {
    localStorage.removeItem('token');
    render(<MobileMenu />);
    expect(screen.getByText('Google konto')).toBeInTheDocument();
  });
  it('renders logout button when logged in', () => {
    localStorage.setItem('token', 'mock-token');
    render(<MobileMenu />);
    expect(screen.getByText('LOGI VÄLJA')).toBeInTheDocument();
    localStorage.removeItem('token');
  });
  it('renders admin functionalities for admin users', () => {
    render(<MobileMenu admin={true} />);
    expect(screen.getByText('KASUTAJATE HALDUS')).toBeInTheDocument();
  });
  it('doesnt render admin functionalities if admin=false', () => {
    render(<MobileMenu admin={false} />);
    expect(screen.queryByText('KASUTAJATE HALDUS')).not.toBeInTheDocument();
  });
  it('handles user login click', () => {
    const mockLogin = jest.fn();
    render(<MobileMenu login={mockLogin} />);
    fireEvent.click(screen.getByText('Google konto'));
    expect(mockLogin).toHaveBeenCalled();
  });
  it('handles user logout click', () => {
    const mockLogOut = jest.fn();
    localStorage.setItem('token', 'mock-token');
    render(<MobileMenu logOut={mockLogOut} />);
    fireEvent.click(screen.getByText('LOGI VÄLJA'));
    expect(mockLogOut).toHaveBeenCalled();
    localStorage.removeItem('token');
  }); */
});