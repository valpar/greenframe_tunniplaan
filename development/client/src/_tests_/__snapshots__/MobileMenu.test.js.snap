// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu matches snapshot 1`] = `
<div
  className="animate-slideRightOff fixed top-20 w-full h-full left-1/2 -translate-x-1/2 bg-white z-30 shadow-lg"
>
  <div
    className="absolute hidden justify-center items-center w-full h-full bg-neutral-900 opacity-70 z-30"
  />
  <div
    className="p-4"
  >
    <div>
      <div
        className="flex flex-col"
      >
        <div
          className="px-4 pt-8"
        >
          <div
            className="flex flex-col space-y-8 w-full pb-10"
          >
            <div
              className="mx-auto w-24 h-24"
            >
              <img
                alt="User"
                className="w-full h-full rounded-full object-cover
 "
              />
            </div>
            <p
              className="uppercase text-center"
            />
            <button
              className="btn-m-menu"
              name="Login Google"
              type="button"
            />
            <a
              className="btn-m-menu text-center"
              href="https://start.hk.tlu.ee/sahtelbeta/sahtel/index.php"
              rel="noopener noreferrer"
              target="_blank"
            >
              Sahtel
            </a>
            <a
              className="btn-m-menu text-center"
              href="https://images.pexels.com/photos/159711/books-bookstore-book-reading-159711.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
              rel="noopener noreferrer"
              target="_blank"
            >
              Riiul
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
